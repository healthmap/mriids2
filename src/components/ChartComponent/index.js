import React from "react";
import { connect } from "react-redux";
import { Chart } from "react-google-charts";

import { ChartContainer } from "../styled-components/ChartContainer";
import ChartTypeButtons from "../ChartTypeButtons";
import { options } from "../../constants/GoogleChartOptions";
import {
  prepareEbolaDataForCharts,
  getCovidDataForCharts,
} from "../../utils/chartDataHelpers";

const ChartComponent = ({
  ebolaData,
  ebolaDataCombined,
  covidData,
  filters,
}) => {
  // // Get the chartData based on the outbreak selected in the filters
  const chartData =
    filters.outbreak === "Ebola Outbreak"
      ? prepareEbolaDataForCharts(ebolaData, ebolaDataCombined, filters)
      : getCovidDataForCharts(covidData, filters);

  const showChartTypeButtons = filters.outbreak === "COVID 19";

  return (
    <ChartContainer>
      {showChartTypeButtons && <ChartTypeButtons />}
      <Chart
        width="100%"
        height="100%"
        chartType="ColumnChart"
        loader={<div>Loading Chart</div>}
        data={chartData}
        options={options}
        legendToggle
      />
    </ChartContainer>
  );
};

const mapStateToProps = (state) => ({
  filters: state.filters,
  ebolaData: state.ebola.ebolaData.data,
  ebolaDataCombined: state.ebola.ebolaDataCombined.data,
  covidData: state.covid.caseCounts.data,
});

export default connect(mapStateToProps)(ChartComponent);
