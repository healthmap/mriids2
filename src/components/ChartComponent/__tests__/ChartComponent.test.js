import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import ChartComponent from "../index";
import renderer from "react-test-renderer";
import "jest-styled-components";

const mockStore = configureStore([thunk]);

describe("Tests for the connected ChartComponent", () => {
  let store;
  let component;

  const initialState = {
    filters: {
      country: "All",
      outbreak: "Ebola Outbreak",
      view: "snapshot",
      projection: false,
      dateRange: { from: new Date(2014, 9, 1), to: new Date(2016, 1, 20) },
    },
    ebola: {
      ebolaData: {
        isFetching: 0,
        data: {},
        error: {},
      },
      ebolaDataCombined: {
        isFetching: 0,
        data: {},
        error: {},
      },
      riskData: {
        isFetching: 0,
        data: {},
        error: {},
      },
    },
  };

  beforeEach(() => {
    store = mockStore(initialState);

    component = renderer.create(
      <Provider store={store}>
        <ChartComponent />
      </Provider>
    );
  });
  test("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
