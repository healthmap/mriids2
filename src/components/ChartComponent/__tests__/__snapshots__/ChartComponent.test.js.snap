// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the connected ChartComponent should render with given state from Redux store 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 28rem;
  overflow: hidden;
}

<div
  className="c0"
>
  <h2>
    Weekly Cases in all Locations
  </h2>
  <div>
    Loading Chart
  </div>
</div>
`;
